src/m365/spo/commands/list/list-label-get.ts

```diff
@@ -79,44 +84,56 @@ class SpoListLabelGetCommand extends SpoCommand {
   }
 
   #initOptionSets(): void {
-    this.optionSets.push(['listId', 'listTitle']);
+    this.optionSets.push(['listId', 'listTitle', 'listUrl']);
   }
 
   public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
-    try {  
+    try {
       if (this.verbose) {
-        const list: string = (args.options.listId ? args.options.listId : args.options.listTitle) as string;
-        logger.logToStderr(`Getting label set on the list ${list} in site at ${args.options.webUrl}...`);
+        logger.logToStderr(`Getting label set on the list ${args.options.listId || args.options.listTitle || args.options.listUrl} in site at ${args.options.webUrl}...`);
       }
 
-      let requestUrl: string = '';
+      let requestUrl: string = `${args.options.webUrl}/_api/web/`;
 
       if (args.options.listId) {
         if (this.debug) {
-          logger.logToStderr(`Retrieving List Url from Id '${args.options.listId}'...`);
+          logger.logToStderr(`Retrieving List from Id '${args.options.listId}'...`);
         }
 
-        requestUrl = `${args.options.webUrl}/_api/web/lists(guid'${formatting.encodeQueryParameter(args.options.listId)}')?$expand=RootFolder&$select=RootFolder`;
+        requestUrl += `lists(guid'${formatting.encodeQueryParameter(args.options.listId)}')?$expand=RootFolder&$select=RootFolder`;
       }
-      else {
+      else if (args.options.listTitle) {
         if (this.debug) {
-          logger.logToStderr(`Retrieving List Url from Title '${args.options.listTitle}'...`);
+          logger.logToStderr(`Retrieving List from Title '${args.options.listTitle}'...`);
         }
 
-        requestUrl = `${args.options.webUrl}/_api/web/lists/GetByTitle('${formatting.encodeQueryParameter(args.options.listTitle as string)}')?$expand=RootFolder&$select=RootFolder`;
+        requestUrl += `lists/GetByTitle('${formatting.encodeQueryParameter(args.options.listTitle as string)}')?$expand=RootFolder&$select=RootFolder`;
       }
 
-      let requestOptions: any = {
-        url: requestUrl,
-        headers: {
-          'accept': 'application/json;odata=nometadata'
-        },
-        responseType: 'json'
-      };
+      let listServerRelativeUrl: string = '';
+
+      if (args.options.listUrl) {
+        if (this.debug) {
+          logger.logToStderr(`Retrieving List from URL '${args.options.listUrl}'...`);
+        }
+
+        listServerRelativeUrl = urlUtil.getServerRelativePath(args.options.webUrl, args.options.listUrl);
+      }
+      else {
+        const requestOptions: any = {
```

Comment:
I know it was like this before so probably it may stay but since we will be refactoring here a bit we might as well get rid of this `any` and use `AxiosRequestConfig` instead

```suggestion
        const requestOptions: AxiosRequestConfig = {
```